docs/control-structures.rst

```diff
@@ -114,10 +114,18 @@ otherwise, the ``value`` option would not be available.
 Due to the fact that the EVM considers a call to a non-existing contract to
 always succeed, Solidity uses the ``extcodesize`` opcode to check that
 the contract that is about to be called actually exists (it contains code)
-and causes an exception if it does not.
+and causes an exception if it does not. This check is skipped if return
+data will be decoded after the call and thus the ABI decoder will catch the
+case of a non-existing contract.
+
 Note that this check is not performed in case of :ref:`low-level calls <address_related>` which
 operate on addresses rather than contract instances.
 
+.. note::
+    Be careful when using high-level calls to precompiled contracts,
```

Comment:
I wanted to suggest that we could link to the page or section about the precompiled contracts but looks like we do not really have a dedicated one. The closest I could find was the [Units and Globally Available Variables > Mathematical and Cryptographic Functions](https://docs.soliditylang.org/en/latest/units-and-global-variables.html#mathematical-and-cryptographic-functions) section but it's not only about precompiles. I think we should really have some place that explains what they are in general.

Reply:
I think it should be in the "introduction to EVM" section.

Reply:
Maybe the "introduction to EVM" is too high-level. I have to think about it further.

Reply:
Looks great!